### DISCUSSION TITLE
Wysokość prowizji od 1 szt. produktu na aukcji

### DISCUSSION THREAD

**Discussion by javaguru-pl** [allegro/allegro-api#5629] [Suggestion - verification]
Witam,

Czy w metodach:
https://api.{environment}/sale/offers
https://api.{environment}/sale/offers/{offerId}
mogliby Państwo dodać pole z informacją o **wysokości prowizji od 1 szt. towaru** (wartość bezwzględna wyrażona w PLN)?
Oczywiście wartość ta zmieniała by się gdyby oferta zmieniła np. status wyróżnienia (dodatkowa prowizja od aukcji wyróżnionych), swoją cenę, etc.
Chodzi o to aby móc uzyskać taką informację dla dowolnej własnej aukcji, która jeszcze nie ma sprzedaży (więc pobranie billingów nie rozwiąże tu problemu).
Jest to funkcjonalność podobna jak w kalkulatorze opłat Allegro, z tymże ta metoda działała by dla istniejących już aukcji.

Z góry dziękuję za odpowiedź.

**Comment by PrzemyslawLukanowski** [comment]
Dziękuję za sugestię. Na tę chwilę pozostawiamy ją w przestrzeni Discussions, tak aby również inni użytkownicy mogli się z nią zapoznać i ew. na nią zagłosować.

**Comment by ZakuroPL** [comment]
Witam. Podepnę się pod temat, aby nie tworzyć osobnego wątku.
Chcę pobrać wszystkie sprzedaże w allegro i obliczyć dla każdego zamówienia dokładny zysk. Do tego celu potrzebuje dokładną kwotę kwotę prowizji allegro per zamówienie. To co się już zorientowałem to nie jest to możliwe w sposób bezpośredni. Tak więc jak to zrobić?
W zapytaniu: https://api.{environment}/pricing/offer-fee-preview
mamy response jak w załączonym obrazku
![Zrzut ekranu 2024-02-04 110803](https://github.com/allegro/allegro-api/assets/72575329/488d8f55-1e0e-411a-83a4-3e130ab61b17)

Pytanie, czy parametr cycleDuration określajacy 240h oznacza, że jest to opłata od datetime.now() i 240h wstecz, czy po prostu jakiś nieokreślony co do sekundy zakres godzin?

Być może jest inny sposób, którego ja nie widzę? Będę wdzięczny za poinstruowanie. Ilość ofert sprzedanych w konkretnym czasie możemy sobie pobrać i jeśli wiem, że np od 1 do ostatniego dnia miesiąca sprzedałem 200 sztuk to chciałbym chociaż pozyskać koszt prowizji w identycznym interwale czasowym, abym mógł sobie chociaż podzielić tą kwotę na te 200sztuk, aby mieć średni koszt prowizji per sztuka.

**Reply by PrzemyslawLukanowski** [reply]
Kwotę prowizji dla konkretnego zamówienia pobierzesz za pomocą [GET /billing/billing-entries](https://developer.allegro.pl/tutorials/jak-sprawdzic-oplaty-nn9DOL5PASX#historia-operacji-billingowych). Dodatkowy parametr, dzięki któremu zwrócimy tylko operacje tego typu, to type.id=SUC. W polu order.id otrzymasz id zamówienia, z którym powiązany jest dany wpis.
Co do pytania o cycleDuration, to jest to liczone jest od momentu wystawienia oferty.
